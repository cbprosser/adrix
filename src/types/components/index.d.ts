export * from './AdvancedSearchForm';
export * from './MTGStaticTypes';
export * from './MultipleCardDisplay';
export * from './SimpleCardSearchBar';
export * from './Symbol';
