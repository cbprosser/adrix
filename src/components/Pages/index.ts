export * from './AdvancedSearch';
export * from './Home';
export * from './Results';
