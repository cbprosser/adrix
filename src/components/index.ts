export * from './AppHeader';
export * from './CardDisplay';
export * from './Pages';
export * from './SimpleCardSearchBar';
