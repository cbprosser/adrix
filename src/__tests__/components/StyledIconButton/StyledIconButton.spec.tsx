import { render } from '@testing-library/react';
import { StyledIconButton } from '../../../components';
import SpeedIcon from '@mui/icons-material/Speed';
import { ButtonProps } from '@mui/material';

const renderButton = (variant?: Exclude<ButtonProps['variant'], 'text'>) =>
  render(
    <StyledIconButton variant={variant}>
      <SpeedIcon />
    </StyledIconButton>,
  );

describe('StyledIconButton Suite', () => {
  it('Should match the snapshot when no variant provided', () => {
    const { asFragment } = renderButton();

    expect(asFragment()).toMatchSnapshot();
  });

  it('Should match the snapshot when contained provided', () => {
    const { asFragment } = renderButton('contained');

    expect(asFragment()).toMatchSnapshot();
  });

  it('Should match the snapshot when outlined provided', () => {
    const { asFragment } = renderButton('outlined');

    expect(asFragment()).toMatchSnapshot();
  });
});
