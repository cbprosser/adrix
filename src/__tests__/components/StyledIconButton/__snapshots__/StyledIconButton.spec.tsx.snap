// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`StyledIconButton Suite Should match the snapshot when contained provided 1`] = `
<DocumentFragment>
  <button
    class="MuiButtonBase-root MuiIconButton-root MuiIconButton-sizeMedium css-0-MuiButtonBase-root-MuiIconButton-root"
    tabindex="0"
    type="button"
    variant="contained"
  >
    <svg
      aria-hidden="true"
      class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-1-MuiSvgIcon-root"
      data-testid="SpeedIcon"
      focusable="false"
      viewBox="0 0 24 24"
    >
      <path
        d="m20.38 8.57-1.23 1.85a8 8 0 0 1-.22 7.58H5.07A8 8 0 0 1 15.58 6.85l1.85-1.23A10 10 0 0 0 3.35 19a2 2 0 0 0 1.72 1h13.85a2 2 0 0 0 1.74-1 10 10 0 0 0-.27-10.44zm-9.79 6.84a2 2 0 0 0 2.83 0l5.66-8.49-8.49 5.66a2 2 0 0 0 0 2.83z"
      />
    </svg>
    <span
      class="MuiTouchRipple-root css-2-MuiTouchRipple-root"
    />
  </button>
</DocumentFragment>
`;

exports[`StyledIconButton Suite Should match the snapshot when no variant provided 1`] = `
<DocumentFragment>
  <button
    class="MuiButtonBase-root MuiIconButton-root MuiIconButton-sizeMedium css-0-MuiButtonBase-root-MuiIconButton-root"
    tabindex="0"
    type="button"
  >
    <svg
      aria-hidden="true"
      class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-1-MuiSvgIcon-root"
      data-testid="SpeedIcon"
      focusable="false"
      viewBox="0 0 24 24"
    >
      <path
        d="m20.38 8.57-1.23 1.85a8 8 0 0 1-.22 7.58H5.07A8 8 0 0 1 15.58 6.85l1.85-1.23A10 10 0 0 0 3.35 19a2 2 0 0 0 1.72 1h13.85a2 2 0 0 0 1.74-1 10 10 0 0 0-.27-10.44zm-9.79 6.84a2 2 0 0 0 2.83 0l5.66-8.49-8.49 5.66a2 2 0 0 0 0 2.83z"
      />
    </svg>
    <span
      class="MuiTouchRipple-root css-2-MuiTouchRipple-root"
    />
  </button>
</DocumentFragment>
`;

exports[`StyledIconButton Suite Should match the snapshot when outlined provided 1`] = `
<DocumentFragment>
  <button
    class="MuiButtonBase-root MuiIconButton-root MuiIconButton-sizeMedium css-0-MuiButtonBase-root-MuiIconButton-root"
    tabindex="0"
    type="button"
    variant="outlined"
  >
    <svg
      aria-hidden="true"
      class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-1-MuiSvgIcon-root"
      data-testid="SpeedIcon"
      focusable="false"
      viewBox="0 0 24 24"
    >
      <path
        d="m20.38 8.57-1.23 1.85a8 8 0 0 1-.22 7.58H5.07A8 8 0 0 1 15.58 6.85l1.85-1.23A10 10 0 0 0 3.35 19a2 2 0 0 0 1.72 1h13.85a2 2 0 0 0 1.74-1 10 10 0 0 0-.27-10.44zm-9.79 6.84a2 2 0 0 0 2.83 0l5.66-8.49-8.49 5.66a2 2 0 0 0 0 2.83z"
      />
    </svg>
    <span
      class="MuiTouchRipple-root css-2-MuiTouchRipple-root"
    />
  </button>
</DocumentFragment>
`;
